generator_process/registrar.py

```diff
@@ -1,13 +1,58 @@
 from .intent import Intent
+from enum import IntEnum
+
+class BackendTarget(IntEnum):
+    """Which generator backend to use"""
+    LOCAL = 0
+    STABILITY_SDK = 1
+
+    def __str__(self):
+        return self.name
+    
+    def init_img_actions(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return ['modify', 'inpaint', 'outpaint']
+            case BackendTarget.STABILITY_SDK:
+                return ['modify', 'inpaint']
+    
+    def inpaint_mask_sources(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return ['alpha', 'prompt']
+            case BackendTarget.STABILITY_SDK:
+                return ['alpha']
+    
+    def color_correction(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return True
+            case BackendTarget.STABILITY_SDK:
+                return False
+    
+    def negative_prompts(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return True
+            case BackendTarget.STABILITY_SDK:
+                return False
+    
+    def upscaling(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return True
+            case BackendTarget.STABILITY_SDK:
+                return False
```

Comment:
Seamless should also have its own backend check since DreamStudio does not support it.